src/firmware/app/control/trajectory_planner.h

```diff
@@ -181,123 +213,177 @@ static void app_trajectory_planner_generateConstArclengthTrajectoryPositions(
- * the trajectory
- */
-void static app_trajectory_planner_generatePositionTrajectoryTimeProfile(
-    PositionTrajectoryElement_t* traj_elements, const float num_segments,
-    const float arc_segment_length, float* velocity_profile);
-
-/***
- *  This function takes in a forwards trajectory annd modifies it in place to become a
- * backwards trajectory NOTE: This function exists to avoid issues outlined in #1322
+ * @pre The path parameters of the variable_time_trajectory are all initialized
+ *
+ * @param path_parameters [in] The path parameters for the trajectory
+ *
+ * @param constant_period_trajectory [in/out] Pre-allocated trajectory that will be
+ * modified to represent a constant interpolation trajectory
  *
- *  TODO: Remove when #1322 is merged.
+ * @param interpolation_period [in] The interpolation period desired of the constant
+ * interpolation period trajectory
  *
- * @param forwards_trajectory This is the trajectory that will be modified in place to
- * become a reverse trajectory
+ * @return Generation status, this can be OK or an error message
  */
-void static app_trajectory_planner_reversePositionTrajectoryDirection(
-    PositionTrajectory_t* forwards_trajectory);
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_generateConstantInterpolationPeriodPositionTrajectory(
+    FirmwareRobotPathParameters_t path_parameters,
+    PositionTrajectory_t* constant_period_trajectory, const float interpolation_period);
 
 /**
- * This function generates a velocity trajectory that corresponds to the time-optimal
- * velocity to follow a specified path. This profile is based on the input position
- * trajectory
+ * This function calculates the robot state at every point along the trajectory (X/Y amd
+ * orientation) along with the physical length of these segments (meters, radians)
  *
- * @param position_trajectory [in] The position trajectory to build the velocity
- * trajectory from
+ * @pre All arrays in the Trajectory must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_SEGMENTS
+ *
+ * @param trajectory [in] The trajectory with pre-allocated arrays to contain trajectory
+ * data
  * @param velocity_trajectory [out] The velocity trajectory that corresponds to the
  * time-optimal velocity to follow a specified path
  */
-void app_trajectory_planner_generateVelocityTrajectory(
-    PositionTrajectory_t* position_trajectory, VelocityTrajectory_t* velocity_trajectory);
+void app_trajectory_planner_generateStatesAndReturnSegmentLengths(
+    PositionTrajectory_t* trajectory,
+    TrajectorySegment_t trajectory_segments[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS]);
+
+/**
+ * Function generates a constant interpolation period trajectory based on the input path
+ * parameters and interpolation period
+ *
+ * @pre path parameter variables must all be initialized
+ *
+ * @pre velocity trajectory element array must be pre-allocated up to
```

Comment:
``` suggestion
 * @pre velocity trajectory element array must be pre-allocated up to
```
I might have missed some other cases, but this seems like a fairly common pattern (missing the underscore) so maybe just do a quick pass to double check me?

Reply:
I only saw 1, but is fixed now 👍